WMIAdventure/backend/WMIAdventure_backend/IngameUsers/signals.py

```diff
@@ -1,20 +1,51 @@
 import random
 
+from django.db.models import F, Q, Min
+
 from IngameUsers.businesslogic.experience.Experience import Experience
 from IngameUsers.businesslogic.skill_points import calculate_skill_points_gain
-from IngameUsers.models import UserProfile, Semester, UserStats
+from IngameUsers.models import UserProfile, Semester, UserStats, UserCard
 from WMIAdventure_backend import settings
+from cards.models import CardInfo
 from users.models import User
 from users.signals import user_registered
 from users.views import UserRegister
 from utils.test_data.create_example_data import get_or_create_example_deck_for_user
 
 
+def _give_all_not_owned_cards_to_user(user_profile) -> int:
+    """
+    Gives all cards with minimal level to user if he doesn't own them.
+
+    TODO: REMOVE THIS IN THE FUTURE. For now all users should have all cards. This function will be removed when
+     there will be some way of gaining cards implemented.
+
+    :return: How many cards were given to user.
+    """
+
+    owned_infos = user_profile.user_cards.annotate(info=F('card__info')).values_list('info', flat=True)
+
+    # If user is not owner of all cards then give him those missing cards with minimal level
+    if owned_infos.count() < CardInfo.objects.count():
+        not_owned_infos = CardInfo.objects.filter(~Q(pk__in=owned_infos))
+        cards_to_give = []
+        for card_info in not_owned_infos:
+            # Retrieving Card related to CardInfo with minimal level
+            min_level = card_info.levels.aggregate(Min('level'))['level__min']
+            cards_to_give.append(card_info.levels.get(level=min_level))
```

Comment:
Tutaj jest problem - miałem w bazie jakieś karty, które nie miały poziomu i tutaj `get` nic nie zwracał. 
Wiem, że to edge case, który raczej nie wystąpi, ale lepiej to obsłużyć